File: scratch/debug_db.py
import os
import psycopg2

db_url = os.environ.get("DATABASE_URL")
if not db_url:
    print("DATABASE_URL not set")
    exit(1)

conn = psycopg2.connect(db_url)
cur = conn.cursor()

print("--- API Groups ---")
cur.execute("SELECT id, name, tenant_id FROM api_groups")
rows = cur.fetchall()
for row in rows:
    print(f"Group: {row[0]} | Name: {row[1]} | Tenant: {row[2]}")

print("\n--- Tenants ---")
cur.execute("SELECT id, name, mcp_client_id FROM tenants")
rows = cur.fetchall()
for row in rows:
    print(f"Tenant: {row[0]} | Name: {row[1]} | MCP Client: {row[2]}")

print("\n--- User Groups ---")
cur.execute("SELECT email, group_id FROM user_groups")
rows = cur.fetchall()
for row in rows:
    print(f"User: {row[0]} | Group: {row[1]}")

cur.close()
conn.close()

File: scratch/debug_db.rs
use tokio_postgres::NoTls;
use std::env;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let db_url = env::var("DATABASE_URL").expect("DATABASE_URL not set");
    let (client, connection) = tokio_postgres::connect(&db_url, NoTls).await?;

    tokio::spawn(async move {
        if let Err(e) = connection.await {
            eprintln!("connection error: {}", e);
        }
    });

    println!("--- API Groups ---");
    let rows = client.query("SELECT id, name, tenant_id FROM api_groups", &[]).await?;
    for row in rows {
        let id: String = row.get(0);
        let name: String = row.get(1);
        let tenant_id: Option<String> = row.get(2);
        println!("Group: {} | Name: {} | Tenant: {:?}", id, name, tenant_id);
    }

    println!("\n--- Tenants ---");
    let rows = client.query("SELECT id, name, mcp_client_id FROM tenants", &[]).await?;
    for row in rows {
        let id: String = row.get(0);
        let name: String = row.get(1);
        let mcp_client_id: Option<String> = row.get(2);
        println!("Tenant: {} | Name: {} | MCP Client: {:?}", id, name, mcp_client_id);
    }

    Ok(())
}
